packages/core-bridge/src/errors.ts

```diff
@@ -0,0 +1,40 @@
+import { IllegalStateError } from '@temporalio/common';
+
+/**
+ * The worker has been shut down
+ */
+export class ShutdownError extends Error {
+  public readonly name = 'ShutdownError';
+}
+
+/**
+ * Thrown after shutdown was requested as a response to a poll function, JS should stop polling
+ * once this error is encountered
+ */
+export class TransportError extends Error {
+  public readonly name = 'TransportError';
+}
+
+/**
+ * Something unexpected happened, considered fatal
+ */
+export class UnexpectedError extends Error {
+  public readonly name = 'UnexpectedError';
+}
+export { IllegalStateError };
+
+export function convertFromNamedError(e: unknown): unknown {
+  if (e instanceof Error) {
```

Comment:
You should probably remove the stack trace from these errors because it won't be right.

Reply:
It should be correct up to the point of entering core bridge, at which point it would have been a native call anyway. That should provide enough context to help users understand issues from their perspective.

But at some point, I thought about copying the stacktrace from the original error, but ended up leaving that aside without giving it a try. Will do some test with this.

Reply:
After experimenting with this, I copied the original stack trace in non-callback exceptions, as it contains meaningful stack trace in user code. Setting to blank in the case of callbacks, as there is nothing useful to extract from there.